pandas/core/indexes/category.py

```diff
@@ -324,20 +324,21 @@ def _reverse_indexer(self):
     @Appender(_index_shared_docs['__contains__'] % _index_doc_kwargs)
     def __contains__(self, key):
         hash(key)
-
-        if self.categories._defer_to_indexing:
-            return key in self.categories
-
-        return key in self.values
+        if isna(key):
+            return self.isna().any()
+        try:
+            loc = self.categories.get_loc(key)
+        except KeyError:
+            return False
+        if is_scalar(loc):
+            return loc in self._engine
+        else:  # if self.categories is IntervalIndex, loc is an array
```

Comment:
can you put a blank line between things, e.g.
```
if isna(...):
     ....

try:
   ....
except:
  ....

if is_scalar(...):
   ...

# no else needed here
return ...
```

Reply:
also can you put comments before each case (not everythin needs a comment), but i find this hard to grok in its current form.